### PR TITLE
[J11] TEST-14 Replaced spock-shaded with spock-all pom artifact.

### PR DESCRIPTION
**Pull request by paouelle**
### Description of the Change
This PR replaces the spock-shaded jar file with a new spock-all artifact.
It also adds test cases for Definalizer.

### Benefits
Reduces the size required to archive the artifacts while keeping all required dependencies together in one pom file that can be referenced as a dependency using `<type>pom</type>`.

### Verification Process
- A passing build should be sufficient.
- Corresponding PR on DDF will validate its use.

### Applicable Issues
Fixes: #14

### CODE REVIEW

**Comment by paouelle** [comment]
build now

File: pom.xml [hunk 1]
```diff
@@ -193,6 +195,17 @@
                 <artifactId>mockito-core</artifactId>
                 <version>${mockito-core.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.codehaus.groovy</groupId>
+                <artifactId>groovy-all</artifactId>
+                <version>${groovy.version}</version>
+                <type>pom</type>
+            </dependency>
+            <dependency>
```
Comment (coyotesqrl) [inline pom.xml]: ❓ Why is this here a second time and without the `type` qualifier? Are we pulling in both the pom and a jar?

Comment (paouelle) [reply pom.xml]: The first one is the standard way to get dependencies defined in the pom file. The second is because this version of `groovy-all` also contains classes. The latest one used for Java 11 is a pure pom and as such, the second one should go away.

File: spock-all/pom.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    <parent>
+        <groupId>org.codice.test</groupId>
+        <artifactId>codice-test</artifactId>
+        <version>0.2-SNAPSHOT</version>
+    </parent>
+
+    <name>Codice Test :: Spock Dependencies</name>
+    <artifactId>spock-all</artifactId>
+    <packaging>pom</packaging>
+
```
Comment (peterhuffer) [inline spock-all/pom.xml]: ❓ Can we add awaitility-groovy in here? Something like

<dependency>
            <groupId>org.awaitility</groupId>
            <artifactId>awaitility-groovy</artifactId>
            <version>${awaitility.version}</version>
            <scope>test</scope>
            <exclusions>
                <exclusion>
                    <artifactId>groovy-all</artifactId>
                    <groupId>org.codehaus.groovy</groupId>
                </exclusion>
            </exclusions>
        </dependency>

**Comment by paouelle** [comment]
build now
